auto_vk_toolkit/src/context_vulkan.cpp

```diff
@@ -442,6 +442,27 @@ namespace avk
 		return sem;
 	}
 
+	avk::semaphore context_vulkan::record_and_submit_with_timeline_semaphore(std::vector<avk::recorded_commands_t> aRecordedCommandsAndSyncInstructions, const avk::queue& aQueue, avk::stage::pipeline_stage_flags aSrcSignalStage, uint64_t aSignalValue, uint64_t aInitialValue, vk::CommandBufferUsageFlags aUsageFlags) {
+		auto sem = create_timeline_semaphore(aInitialValue);
+		record_and_submit_with_timeline_semaphore(aRecordedCommandsAndSyncInstructions, aQueue, aSrcSignalStage >> sem | aSignalValue, aUsageFlags);
+		return sem;
+	}
+
+	void context_vulkan::record_and_submit_with_timeline_semaphore(std::vector<avk::recorded_commands_t> aRecordedCommandsAndSyncInstructions, const avk::queue& aQueue, avk::semaphore_signal_info aSignalInfo, vk::CommandBufferUsageFlags aUsageFlags) {
+		auto& cmdPool = get_command_pool_for_single_use_command_buffers(aQueue);
+		auto cmdBfr = cmdPool->alloc_command_buffer(aUsageFlags);
+		
+
+		record(std::move(aRecordedCommandsAndSyncInstructions))
+			.into_command_buffer(cmdBfr)
+			.then_submit_to(aQueue)
+			.signaling_upon_completion(aSignalInfo)
+			.submit();
+
+		aSignalInfo.mSignalSemaphore.get().handle_lifetime_of(std::move(cmdBfr));
+		// TODO we probably need another way to handle the lifetime of resources since timeline semaphores aren't single use ==> live longer
```

Comment:
What I was talking about here was that since timeline semaphores are no longer single-shot synchronization primitives (1 signal, 1 wait) they might have a lifetime close to the whole application runtime. If a command buffer now gets enqueued and executed early on in the semaphore's lifespan, won't it be kept alive until the semaphore is destroyed, way longer than it needs to exist?

Reply:
With the most recent changes, I added an optional `value` parameter to `semaphore_t::handle_lifetime_of`. Here I just pass the same value the semaphore will be signaled to, once the work has finished.

I suspect this is wrong in cases where the passed `aSignalInfo` uses a stage other than `all_commands`. Do you think we should add another param for a cmdBuffer deletion value, or just let the caller setup the lifetime handling themselves?